.github/workflows/vaccine.yml

```diff
@@ -57,19 +57,17 @@ jobs:
       - uses: ruby/setup-ruby@v1
         with:
           ruby-version: .ruby-version
-      - run: bundle install
+      - run: bundle install 
       # Prepare mock server 'catadog', https://github.com/DataDog/catadog
-      - uses: actions/checkout@v4
-        with:
-          repository: datadog/catadog
-          ref: tonycthsu/daemon
-          path: catadog
       - run: mkdir $GITHUB_WORKSPACE/records
       - run: |
-          cd catadog
-          bundle install
-          bundle exec catadog -h 0.0.0.0 --mock :Sink --no-forward --daemon --record $GITHUB_WORKSPACE/records
-      - run: curl -I http://localhost:8128/catadog
+          docker run -d \
+            --name catadog \
+            -v $GITHUB_WORKSPACE/records:/records \
+            -p 8128:8128 \
+            ghcr.io/datadog/catadog:latest \
+            "-h 0.0.0.0 --mock :Sink --no-forward --record /records"
+      - run: curl -I --retry 12 -f --retry-all-errors --retry-delay 5 -s http://localhost:8128/catadog
```

Comment:
`--retry 12 -f --retry-all-errors --retry-delay 5 -s` was added to the original `curl` command. Without retrying and with only running the `curl` command once, I was getting the following error: `curl: (56) Recv failure: Connection reset by peer`. The retry is helpful in giving the Docker container enough time to start up; however, is this the best solution?

Reply:
There are other mechanism to wait for a container to be ready, but this is good enough

Reply:
@sarahchen6 The idiomatic way would be to use `HEALTHCHECK`, see the [`Dockerfile` reference](https://docs.docker.com/reference/dockerfile/).

In that context, using `http://localhost:8128/catadog` is good enough for this as well, but the idiomatic way is to have a dedicated healthcheck endpoint, conventionally [a `healthz` wording is used]( https://stackoverflow.com/questions/43380939/where-does-the-convention-of-using-healthz-for-application-health-checks-come-f), so ``http://localhost:8128/catadog/healthz` would be fine.